server-spi-private/src/main/java/org/keycloak/models/utils/RepresentationToModel.java

```diff
@@ -1406,6 +1408,11 @@ private static ClientModel createClient(KeycloakSession session, RealmModel real
             }
         }
 
+        if ("saml".equals(resourceRep.getProtocol())
```

Comment:
Prefer `SamlProtocol.LOGIN_PROTOCOL` (other places as well)

Reply:
`SamlProtocol` is part of services and `RepresentationToModel` is in `server-spi-private` so it is not available there. I can create some constants class in `server-spi-private` and then reuse it everywhere, but I am not sure this is what we want.

Reply:
Ah, that makes perfect sense :)